frontend/src/Components/Shared/TreeManager/TreeManager.spec.tsx

```diff
@@ -1,21 +1,461 @@
+      expect(iconSpanNode2).toHaveClass('anticon-setting');
+    });
+    // TODO: how to test for FontAwesome?
+  });
+
+  it('renders the node title', async () => {
+    const { container } = getComponent();
+    await waitFor(() => {
+      const nodeTitles = container.querySelectorAll('.tree-node-title');
+      expect(nodeTitles.length).toBe(2);
+      expect(nodeTitles.item(0)?.textContent).toBe('node1');
+      expect(nodeTitles.item(1)?.textContent).toBe('leaf1');
+    });
+  });
+
+  it('renders the node options icon', async() => {
+    const { container } = getComponent();
+    await waitFor(() => {
+      const nodeSettingsIcon = container.querySelectorAll('.tree-manager-node-actions');
+      expect(nodeSettingsIcon.length).toBe(2);
+      const iconSpanNode1 = nodeSettingsIcon.item(0)?.querySelector('span');
+      const iconSpanNode2 = nodeSettingsIcon.item(1)?.querySelector('span');
+      expect(iconSpanNode1).toHaveClass('anticon-setting');
+      expect(iconSpanNode2).toHaveClass('anticon-setting');
+    });
+  });
+
+  it('renders the context menu when right clicking the node title if contextMenuOnNode prop is true', async() => {
+    const { container } = getComponent({ ...requiredProps, contextMenuOnNode: true });
+    const nodeTitles = container.querySelectorAll('.tree-manager-node-title');
+    const titleNode1 = nodeTitles.item(0);
+    // right mouse click
+    expect(screen.queryByText('Add new layer group')).toBeNull(); 
+    fireEvent.contextMenu(titleNode1);
+    await waitFor(() => {
+      expect(screen.queryByText('Add new layer group')).toBeDefined();
+    });
+  });
+
+  it('does not render the context menu when right clicking the node title if ' +
+  'contextMenuOnNode prop is false', async() => {
+    // contextMenuOnNode is false by default
+    const { container } = getComponent();
+    const nodeTitles = container.querySelectorAll('.tree-manager-node-title');
+    const firstNodeTitle = nodeTitles.item(0);
+    // right mouse click
+    fireEvent.contextMenu(firstNodeTitle);
+    await waitFor(() => {
+      const contextMenu = screen.queryByText('Add new layer group');
+      expect(contextMenu).toBe(null);
+    });
+  });
+
+  it('renders the context menu when clicking the options icon', async() => {
+    const { container } = getComponent();
+    const nodeSettings = container.querySelectorAll('.tree-manager-node-actions');
+    const settingsNode1 = nodeSettings.item(0)?.querySelector('button');
+    const settingsNode2 = nodeSettings.item(1)?.querySelector('button');
+    expect(screen.queryByText('Add new layer group')).toBeNull(); 
+    fireEvent.click(settingsNode1 as Element);
+    fireEvent.click(settingsNode2 as Element);
+    await waitFor(() => {
+      expect(screen.queryByText('Add new layer group')).toBeDefined();
+    });
+  });
+
+  it('renders a modal when clicking to add a new node on the options menu and attributeContainer ' +
```

Comment:
we don't need the modal in our application, so shouldn't we remove it?
YAGNI: https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it

Reply:
I have remove the modal from the Tree Manager component as well